### PR TITLE
fix: classes as body parameters are always Object

### LINKED ISSUE #61: Classes that are used as body parameters are not resolved and always result in a parameter of type "Object"
The JSON below produces the following dart method:

```dart
  @PUT('/users/credentials')
  Future<void> changeUserCredentials({
    @Body() required Object credentials,
  });
```

while the expected content is
```dart
  @PUT('/users/credentials')
  Future<void> changeUserCredentials({
    @Body() required Credentials credentials,
  });
```

JSON:
```json
{
  "swagger": "2.0",
  "info": {
    "version": "1.0",
    "title": "User Service"
  },
  "host": "",
  "basePath": "/api",
  "tags": [
    {
      "name": "Users",
      "description": "User Controller"
    }
  ],
  "paths": {
    "/users/credentials": {
      "put": {
        "tags": [
          "Users"
        ],
        "summary": "Changes the user's credentials",
        "operationId": "changeUserCredentials",
        "consumes": [
          "application/json"
        ],
        "produces": [
          "*/*"
        ],
        "parameters": [
          {
            "in": "body",
            "name": "credentials",
            "description": "credentials",
            "required": true,
            "schema": {
              "$ref": "#/definitions/Credentials"
            }
          }
        ],
        "responses": {
          "204": {
            "description": "Credentials changed correctly"
          }
        },
        "deprecated": false
      }
    }
  },
  "definitions": {
    "Credentials": {
      "type": "object",
      "required": [
        "email",
        "password"
      ],
      "properties": {
        "email": {
          "type": "string",
          "description": "The new email"
        },
        "password": {
          "type": "string"
        }
      },
      "title": "Credentials"
    }
  }
}

```

### PR DESCRIPTION
**Pull request by elenaferr0**
fix  #61

### CODE REVIEW

File: swagger_parser/lib/src/parser/parser.dart [hunk 1]
```diff
@@ -143,13 +143,25 @@ class OpenApiParser {
...
+            );
+          } else {
+            typeWithImport = _findType(
+              rawParameter,
+              name: rawParameter[_nameConst].toString(),
+              isRequired: isRequired ?? true,
+              useSchema: true,
+            );
+          }
+
```
Comment (Carapacik) [inline swagger_parser/lib/src/parser/parser.dart:164]: Maybe 
```dart
        final typeWithImport = _findType(
            rawParameter[_schemaConst] != null ? rawParameter[_schemaConst] as Map<String, dynamic> : rawParameter,
            name: rawParameter[_nameConst].toString(),
            isRequired: isRequired ?? true,
            allOfObject: (rawParameter[_schemaConst] as Map<String, dynamic>)
                .containsKey(_allOfConst),
          );
```

File: swagger_parser/lib/src/parser/parser.dart [hunk 2]
```diff
@@ -319,12 +331,25 @@ class OpenApiParser {
...
+        } else {
+          typeWithImport = _findType(
+            rawParameter,
+            name: rawParameter[_nameConst].toString(),
+            isRequired: isRequired ?? true,
+            useSchema: true,
+          );
+        }
+
         if (typeWithImport.import != null) {
```
Comment (Carapacik) [inline swagger_parser/lib/src/parser/parser.dart:353]: Maybe 
```dart
        final typeWithImport = _findType(
            rawParameter[_schemaConst] != null ? rawParameter[_schemaConst] as Map<String, dynamic> : rawParameter,
            name: rawParameter[_nameConst].toString(),
            isRequired: isRequired ?? true,
            allOfObject: (rawParameter[_schemaConst] as Map<String, dynamic>)
                .containsKey(_allOfConst),
          );
```

File: swagger_parser/lib/src/parser/parser.dart (hunk 1 above)
Comment (elenaferr0) [reply swagger_parser/lib/src/parser/parser.dart]: done

File: swagger_parser/lib/src/parser/parser.dart (hunk 2 above)
Comment (elenaferr0) [reply swagger_parser/lib/src/parser/parser.dart]: done

**Comment by Carapacik** [comment]
Thanks

**Pull request merged**